chemprop/v2/utils/v1_to_v2.py

```diff
@@ -0,0 +1,140 @@
+import torch
+from chemprop.v2.models.metrics import MetricRegistry
+from chemprop.v2.models.modules.agg import AggregationRegistry
+from chemprop.v2.models.modules.readout import ReadoutRegistry
+from chemprop.v2.models.loss import LossFunctionRegistry
+from chemprop.v2.models.modules.message_passing.molecule import BondMessageBlock, AtomMessageBlock
+
+from lightning.pytorch.utilities.parsing import AttributeDict
+from lightning.pytorch import __version__
+
+def convert_state_dict_v1_to_v2(model_v1: dict) -> dict:
+    """Converts v1 model dictionary to a v2 state dictionary
+    """
+
+    state_dict_v2 = {}
+
+    if model_v1["args"].target_weights is not None:
+        state_dict_v2["w_t"] = torch.tensor(model_v1["args"].target_weights).unsqueeze(0)
+    else:
+        state_dict_v2["w_t"] = torch.ones(model_v1["args"].num_tasks).unsqueeze(0)
+
+    state_dict_v1 = model_v1["state_dict"]
+    state_dict_v2["message_passing.W_i.weight"] = state_dict_v1[
+        "encoder.encoder.0.W_i.weight"
+    ]
+    state_dict_v2["message_passing.W_h.weight"] = state_dict_v1[
+        "encoder.encoder.0.W_h.weight"
+    ]
+    state_dict_v2["message_passing.W_o.weight"] = state_dict_v1[
+        "encoder.encoder.0.W_o.weight"
+    ]
+    state_dict_v2["message_passing.W_o.bias"] = state_dict_v1[
+        "encoder.encoder.0.W_o.bias"
+    ]
+
+    state_dict_v2["readout.loc"] = torch.from_numpy(
+        model_v1["data_scaler"]["means"]
+    ).unsqueeze(0)
+    state_dict_v2["readout.scale"] = torch.from_numpy(
+        model_v1["data_scaler"]["stds"]
+    ).unsqueeze(0)
+
+    for i in range(model_v1["args"].ffn_num_layers):
+        state_dict_v2[f"readout.ffn.ffn.{i*3}.weight"] = state_dict_v1[
+            f"readout.{i*3+1}.weight"
+        ]
+        state_dict_v2[f"readout.ffn.ffn.{i*3}.bias"] = state_dict_v1[
+            f"readout.{i*3+1}.bias"
+        ]
+
+    return state_dict_v2
+
+
+def convert_hyper_parameters_v1_to_v2(model_v1: dict) -> dict:
+    """Converts v1 model dictionary to v2 hyper_parameters dictionary
+    """
+    hyper_parameters_v2 = {}
+
+    args_v1 = model_v1["args"]
+    hyper_parameters_v2[
+        "batch_norm"
+    ] = False  # It doesn't seem that batch norm is done in v1
```

Comment:
yes, batch norm is new in v2

Reply:
I think you could remove the comment or change it to " # Batch norm is new to v2". My preference is remove.
Also I don't know why this hyperparameter is spread over three lines. I think it could be collapsed to just one.
`hyper_parameters_v2["batch_norm"] = False`

Reply:
Thanks! I remove the comment in the new push. I use `black` as my default formatter, and it tries to format the code if the code is too long to improve readability. In the new push I also make this into one line.